### DISCUSSION TITLE
Update auf Knopfdruck oder sogar automatisch

### DISCUSSION THREAD

**Discussion by themanfrommoon** [tbnobody/OpenDTU#1597] [Ideas]
Moin,

leider muss man zum Updaten immer die binary Datei runterladen und dann auf allen DTUs (ich habe drei Stück im Betrieb) manuell eine Firmwareaktualisierung durchführen.

Cooler wäre es doch, wenn das auf Knopfdruck ....oder sogar automatisch funktionieren würde.

Auf Knopfdruck geht es z.B. bei tasmota.
Da sieht das so aus:
![grafik](https://github.com/tbnobody/OpenDTU/assets/28815063/95148d24-ac2e-44bb-811d-689b9b4acfd9)
Die OTA URL ist:
http://ota.tasmota.com/tasmota/release/tasmota-DE.bin.gz

Ein Klick auf "Update starten" und der Rest läuft automatisch.

Ausserdem könnte man auf der Hauptansicht in der oberen Zeile einen Hinweis anzeigen lassen, wenn eine neue Softwareversion verfügbar ist:
![grafik](https://github.com/tbnobody/OpenDTU/assets/28815063/3736425c-bc42-4907-816a-465d5699fc7e)

Für die ganz Mutingen (ich wäre dabei) würde ich sogar eine Option cool finden, die Updates automatisch einspielt, wenn sie verfügbar sind. Einmal am Tag (natürlich Nachts, damit die Tagesstatistik nicht gelöscht wird) nach Updates suchen, und wenn es eine neuere Version gibt automatisch installieren.

Besten Dank und viele Grüße,
Chris

**Comment by themanfrommoon** [comment]
ich push mich mal selbst nach oben....bin vermutlich übersehen worden?!

**Reply by frank-tanne** [reply]
Das braucht doch keiner. Ich möchte nicht jedes Update automatisch haben.
Und das runterladen ist nun auch keine Problem.

**Reply by ituri** [reply]
Der Vorschlag von @themanfrommoon war ja in erster Linie nicht, dass die Updates automatisch eingespielt werden, sondern dass sich das Update ohne separaten Download per Knopfdruck durchführen lässt. 

Letzteres begrüße ich sehr, würde es doch insbesondere die Updates vom Smartphone aus deutlich einfacher gestalten.

**Reply by themanfrommoon** [reply]
....ich brauche die letzten 27 Updates auch nicht, es hat bei mir auch vorher funktioniert.

Mein Vorschlag erhöht den Komfort, nix anderes. Offenbar hat @frank-tanne nicht richtig gelesen...oder nicht richtig verstanden. Der Automatismus ist als Option gedacht. Ich habe bei meinen FritzBoxen und Repeatern (8 Stück!) auch die automatischen Updates eingestellt, und fahre damit seit Jahrzehnten sehr gut.

Beste Grüße,
Chris

**Comment by Tiese** [comment]
Ich habe, bei meinem OpenDTU-Client ([Info](https://github.com/tbnobody/OpenDTU/discussions/1523)), es so programmiert, dass **_im Display_** eine Zeile erscheint - mit der Meldung: "Firmware veraltet".
Und im Browserfenster steht dann - **im Startbildschirm** - hinter der Firmware-Version "veraltet" hinterlegt mit dem **_Link zum direkten Download_**!

Nur mal so am Rande ;-)

**Comment by MetaChuh** [comment]
@themanfrommoon
a one click update will cause a flood of issues, especially if something vital changes between versions, and people click on an update link blindly, even tagging "i have read and understood all changes"

true that to some it's a lack of convenience, but the way it is also ensures a "never change a running system" stability at specifically their places.
(keep in mind, that many older people got their solar installations from their kids or their grandchildren, and imagine if grandparents click on a button, just because it's there 😉)

everyone else, who wants to try out new stuff, and is aware of the risks, should imho not have any issues with more than one step for a free upgrade, as well as having a bit more awareness, depending on their current entoxication level.

**Comment by themanfrommoon** [comment]
Ähm, ja, man kann natürlich immer einfach gegen alles sein und ständig überall nur schwarz sehen und sich überlegen warum man etwas nicht so toll findet, oder man schaut sich mal an, wie andere es machen und sich überlegen warum man das gut findet. Siehe Eingangspost.
Meinetwegen kann man ja auch noch eine einschaltbare Updatesperre mit Passwortschutz einbauen.
Ich habe leide rnicht mehr mitgezählt, wie viele Updates ich im letzten dreiviertel Jahr eingespielt habe, aber es waren reichlich. Und bei keinem Update hatte ich ein Problem.

Wenn ein "one-click"-update so problematisch wäre, wie gerade zuvor beschrieben wurde, dann gäbe es ja sehr viele Probleme bei tasmota.
Das Gegenteil ist der Fall: Bei tasmota funktioniert das Update einfach und komfortabel.

Wo ist der Unterschied, wenn man umständlich eine Firmware runterladen muss, und die dann über ein extra Menü die Updatedatei auswählen muss, um das Update durchzuführen? ...im Zeitaufwand!
Der Hemmschwelle kann man bei Bedarf ein Passwort vorschieben. Das mag ja für den kleineren Teil der User sinnvoll sein.

Also ich finde nach wie vor ein Update, so wie z.B. in tasmota oben gezeigt sehr hilfreich und würde mich freuen, wenn es das hier so oder in ähnlicher Form auch geben würde :-)

Beste Grüße,
Chris

**Reply by frank-tanne** [reply]
Die Entwickler der opendtu machen das ja alles umsonst ,da kann ich mir gut vorstellen das nicht jeder wunsch erfüllt werden kann.
Im übrigen kannst du dich ja mal selbst versuchen
